### ISSUE TITLE
🐛 Bug Report: ui:options not respected in templates

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#18493]
### 📜 Description

Trying to use ui:options in scaffolder templates is not working. For example `inputType` should be supported by the react-json-schema-form: https://rjsf-team.github.io/react-jsonschema-form/docs/api-reference/uiSchema/#inputtype

### 👍 Expected behavior

```html
<input aria-invalid="false" id="root_emailInput" placeholder="placeholder@test.org" type="email" class="MuiInputBase-input-3333 MuiInput-input-3321" value="">
```

### 👎 Actual Behavior with Screenshots

```html
<input aria-invalid="false" id="root_emailInput" placeholder="placeholder@test.org" type="text" class="MuiInputBase-input-3333 MuiInput-input-3321" value="">
```

### 👟 Reproduction steps

Create template with following input:

```yaml
        emailInput:
          title: Email input
          type: string
          ui:placeholder: placeholder@test.org
          ui:options:
            inputType: email
```

### 🖥️ Your Environment

OS:   Darwin 22.5.0 - darwin/arm64
node: v16.20.0
yarn: 1.22.19
cli:  0.22.8 (installed)
backstage:  1.15.0

Dependencies:
  @backstage/app-defaults                                   1.4.0
  @backstage/backend-app-api                                0.4.4
  @backstage/backend-common                                 0.17.0, 0.18.5, 0.19.0
  @backstage/backend-dev-utils                              0.1.1
  @backstage/backend-plugin-api                             0.5.3
  @backstage/backend-tasks                                  0.5.3
  @backstage/backend-test-utils                             0.1.38
  @backstage/catalog-client                                 1.4.2
  @backstage/catalog-model                                  1.4.0
  @backstage/cli-common                                     0.1.12
  @backstage/cli-node                                       0.1.1
  @backstage/cli                                            0.22.8
  @backstage/config-loader                                  1.3.1
  @backstage/config                                         1.0.8
  @backstage/core-app-api                                   1.8.1
  @backstage/core-components                                0.12.5, 0.13.2
  @backstage/core-plugin-api                                1.5.2
  @backstage/dev-utils                                      1.0.16
  @backstage/errors                                         1.2.0
  @backstage/eslint-plugin                                  0.1.3
  @backstage/integration-aws-node                           0.1.4
  @backstage/integration-react                              1.1.14
  @backstage/integration                                    1.5.0
  @backstage/plugin-api-docs                                0.9.5
  @backstage/plugin-app-backend                             0.3.46
  @backstage/plugin-auth-backend                            0.18.4
  @backstage/plugin-auth-node                               0.2.15
  @backstage/plugin-catalog-backend-module-aws              0.2.1
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.1.11
  @backstage/plugin-catalog-backend-module-msgraph          0.5.5
  @backstage/plugin-catalog-backend                         1.10.0
  @backstage/plugin-catalog-common                          1.0.14
  @backstage/plugin-catalog-graph                           0.2.31
  @backstage/plugin-catalog-node                            1.3.7
  @backstage/plugin-catalog-react                           1.7.0
  @backstage/plugin-catalog                                 1.11.2
  @backstage/plugin-devtools-backend                        0.1.1
  @backstage/plugin-devtools-common                         0.1.1
  @backstage/plugin-devtools                                0.1.1
  @backstage/plugin-entity-feedback-backend                 0.1.4
  @backstage/plugin-entity-feedback-common                  0.1.1
  @backstage/plugin-entity-feedback                         0.2.2
  @backstage/plugin-events-node                             0.2.7
  @backstage/plugin-github-actions                          0.6.0
  @backstage/plugin-home-react                              0.1.0
  @backstage/plugin-home                                    0.5.3
  @backstage/plugin-jenkins-backend                         0.2.1
  @backstage/plugin-jenkins-common                          0.1.16
  @backstage/plugin-jenkins                                 0.8.1
  @backstage/plugin-kubernetes-common                       0.6.4
  @backstage/plugin-microsoft-calendar                      0.1.4
  @backstage/plugin-org                                     0.6.9
  @backstage/plugin-permission-backend                      0.5.21
  @backstage/plugin-permission-common                       0.7.6
  @backstage/plugin-permission-node                         0.7.9
  @backstage/plugin-permission-react                        0.4.13
  @backstage/plugin-proxy-backend                           0.2.40
  @backstage/plugin-scaffolder-backend-module-cookiecutter  0.2.22
  @backstage/plugin-scaffolder-backend                      1.15.0
  @backstage/plugin-scaffolder-common                       1.3.1
  @backstage/plugin-scaffolder-node                         0.1.4
  @backstage/plugin-scaffolder-react                        1.5.0
  @backstage/plugin-scaffolder                              1.14.0
  @backstage/plugin-search-backend-module-catalog           0.1.2
  @backstage/plugin-search-backend-module-elasticsearch     1.3.1
  @backstage/plugin-search-backend-module-pg                0.5.7
  @backstage/plugin-search-backend-module-techdocs          0.1.2
  @backstage/plugin-search-backend-node                     1.2.2
  @backstage/plugin-search-backend                          1.3.2
  @backstage/plugin-search-common                           1.2.4
  @backstage/plugin-search-react                            1.6.1, 1.6.2
  @backstage/plugin-search                                  1.3.2
  @backstage/plugin-sonarqube-backend                       0.2.0
  @backstage/plugin-sonarqube-react                         0.1.6
  @backstage/plugin-sonarqube                               0.7.0
  @backstage/plugin-techdocs-backend                        1.6.3
  @backstage/plugin-techdocs-module-addons-contrib          1.0.14
  @backstage/plugin-techdocs-node                           1.7.2
  @backstage/plugin-techdocs-react                          1.1.7
  @backstage/plugin-techdocs                                1.6.4
  @backstage/plugin-user-settings                           0.7.4
  @backstage/release-manifests                              0.0.9
  @backstage/test-utils                                     1.4.0
  @backstage/theme                                          0.2.19, 0.3.0, 0.4.0
  @backstage/types                                          1.1.0
  @backstage/version-bridge                                 1.0.4

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by npiyush97** [comment]
https://github.com/rjsf-team/react-jsonschema-form/issues/2057

**Comment by npiyush97** [comment]
@drodil fix in #20221

**Comment by camilaibs** [comment]
Closed by https://github.com/backstage/backstage/pull/20221
